cmd/ipfs/init.go

```diff
@@ -102,8 +112,22 @@ environment variable:
 			}
 		}
 
+		cfgLocation, _ := req.Options[configOptionName].(string)
 		profile, _ := req.Options[profileOptionName].(string)
 
+		if cfgLocation != "" {
+			if profile != "" {
+				return errInitConfigArgs
```

Comment:
Any reason to reject this?

Reply:
Probably not.
I was thinking we needed to merge profiles rather than use the config transformers. I'll update this.

Reply:
Okay so this works now.
```CMD
set IPFS_PATH=Z:\ipfs-dev-tmp1
ipfs init
set IPFS_PATH=Z:\ipfs-dev-tmp2
ipfs daemon --init --init-config-file=Z:\ipfs-dev-tmp1\config -profile=badgerds
diff Z:\ipfs-dev-tmp1 Z:\ipfs-dev-tmp2
```
```diff

Only in Z:\ipfs-dev-tmp2: badgerds
Only in Z:\ipfs-dev-tmp1: blocks
diff "Z:\\ipfs-dev-tmp1/config" "Z:\\ipfs-dev-tmp2/config"
11,34c11,18
<       "mounts": [
<         {
<           "child": {
<             "path": "blocks",
<             "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
<             "sync": true,
<             "type": "flatfs"
<           },
<           "mountpoint": "/blocks",
<           "prefix": "flatfs.datastore",
<           "type": "measure"
<         },
<         {
<           "child": {
<             "compression": "none",
<             "path": "datastore",
<             "type": "levelds"
<           },
<           "mountpoint": "/",
<           "prefix": "leveldb.datastore",
<           "type": "measure"
<         }
<       ],
<       "type": "mount"
---
>       "child": {
>         "path": "badgerds",
>         "syncWrites": true,
>         "truncate": true,
>         "type": "badgerds"
>       },
>       "prefix": "badger.datastore",
>       "type": "measure"
Only in Z:\ipfs-dev-tmp1: datastore
diff "Z:\\ipfs-dev-tmp1/datastore_spec" "Z:\\ipfs-dev-tmp2/datastore_spec"
1c1
< {"mounts":[{"mountpoint":"/blocks","path":"blocks","shardFunc":"/repo/flatfs/shard/v1/next-to-last/2","type":"flatfs"},{"mountpoint":"/","path":"datastore","type":"levelds"}],"type":"mount"}
\ No newline at end of file
---
> {"path":"badgerds","type":"badgerds"}
\ No newline at end of file
Common subdirectories: Z:\ipfs-dev-tmp1/keystore and Z:\ipfs-dev-tmp2/keystore
```

Reply:
I'd say allow both (but we can stick to transformers for now). That is:

1. Take the initial config, if supplied.
2. Merge it with the default config (later).
3. Apply transformers (if specified).